grpc-sys/build.rs

```diff
@@ -249,40 +250,56 @@ fn bindgen_grpc(mut config: bindgen::Builder) {
         file.read_to_string(&mut buf)
             .expect("Coundn't read header content");
         if buf.contains("GRPCAPI") || buf.contains("GPRAPI") {
-            config = config.header(dent.path().to_str().unwrap());
+            headers.push(String::from(dent.path().to_str().unwrap()));
         }
     }
 
-    config = config.header("grpc_wrap.cc");
+    // To control the order of bindings
+    headers.sort();
+    for path in headers {
+        config = config.header(path);
+    }
 
-    let bindings = config
+    config
+        .header("grpc_wrap.cc")
         .clang_arg("-xc++")
         .clang_arg("-I./grpc/include")
         .clang_arg("-std=c++11")
+        .whitelist_recursively(false)
         .whitelist_function(r"\bgrpc_.*")
         .whitelist_function(r"\bgpr_.*")
         .whitelist_function(r"\bgrpcwrap_.*")
         .whitelist_var(r"\bGRPC_.*")
+        .whitelist_type(r"\bgrpc_.*")
+        .whitelist_type(r"\bgpr_.*")
+        .whitelist_type(r"\bgrpcwrap_.*")
+        .whitelist_type(r"\bcensus_context.*")
+        .whitelist_type(r"\bverify_peer_options.*")
+        .blacklist_function(r"\bgpr_mu_.*")
+        .blacklist_function(r"\bgpr_cv_.*")
+        .blacklist_function(r"\bgpr_once_.*")
+        .blacklist_type(r"gpr_mu")
+        .blacklist_type(r"gpr_cv")
+        .blacklist_type(r"gpr_once")
+        .constified_enum_module(r"grpc_status_code")
+        .no_copy("grpc_slice")
         .default_enum_style(bindgen::EnumVariation::Rust {
             non_exhaustive: false,
         })
-        .constified_enum_module(r"grpc_status_code")
-        .no_copy("grpc_slice")
         .generate()
-        .expect("Unable to generate grpc bindings");
-
-    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-    bindings
-        .write_to_file(out_path.join("grpc-bindings.rs"))
+        .expect("Unable to generate grpc bindings")
+        .write_to_file(file_path)
         .expect("Couldn't write bindings!");
 }
 
 fn main() {
-    let mut cc = Build::new();
-    let mut bind_config = bindgen::Builder::default();
-
     println!("cargo:rerun-if-changed=grpc_wrap.cc");
     println!("cargo:rerun-if-changed=grpc");
+    println!("cargo:rerun-if-changed=bindings");
```

Comment:
I don't think it's necessary.

Reply:
For example, when you accidentally changed the file here.  You find that everything hasn't changed since you ran the script.  You thought that the binding was generated, but it is not.

Reply:
Good point! But is it working as expected? As only the modified timestamp of directory bindings instead of actual bindings files is considered.

Reply:
The timestamp of the file affects the timestamp of the directory, and this is as expected

Reply:
I don't think so.
```
/tmp % ls -alh test-dir
total 0
drwxrwxr-x.  2 jay  jay   60 Aug  5 14:56 .
drwxrwxrwt. 22 root root 520 Aug  5 14:56 ..
-rw-rw-r--.  1 jay  jay    0 Aug  5 14:56 testfile
/tmp % touch test-dir/testfile 
/tmp % ls -alh test-dir       
total 0
drwxrwxr-x.  2 jay  jay   60 Aug  5 14:56 .
drwxrwxrwt. 22 root root 520 Aug  5 14:56 ..
-rw-rw-r--.  1 jay  jay    0 Aug  5 14:57 testfile
```